wpilibc/src/main/native/cpp/event/BooleanEvent.cpp

```diff
@@ -7,42 +7,48 @@
 using namespace frc;
 
 BooleanEvent::BooleanEvent(EventLoop* loop, std::function<bool()> condition)
-    : m_loop(loop), m_condition(std::move(condition)) {}
+    : m_loop(loop),
+      m_condition(std::make_shared<std::function<bool()>>(condition)) {
+  m_state = std::make_shared<bool>((*m_condition)());
+  m_loop->Bind([condition = *m_condition, state = m_state] {  // NOLINT
```

Comment:
Include the warning type so we don't forget what the NOLINT is suppressing.
```suggestion
  m_loop->Bind([condition = *m_condition, state = m_state] {  // NOLINT(clang-analyzer-cplusplus.NewDeleteLeaks)
```

Reply:
Thank you!